docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,591 @@
+# Writing Libraries with Cross Platform Tools
```

Comment:
I am wondering if there is a simpler flow:

- Targeting .NET Core.
- Targeting PCL
- Targeting .NET Framework
- Cross compiling

The current doc is a mix ... meaning that PCL includes 'dotnet' and .NET Framework includes 'dotnet'. That's not strictly needed. We don't want people to think it is required to always target 'dotnet'.